tests/media_thumbnail_test.go

```diff
@@ -68,6 +72,60 @@ func TestRemotePngThumbnail(t *testing.T) {
 	}
 }
 
+func TestFederationThumbnail(t *testing.T) {
+	runtime.SkipIf(t, runtime.Dendrite)
+
+	deployment := complement.Deploy(t, 1)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+
+	srv := federation.NewServer(t, deployment,
+		federation.HandleKeyRequests(),
+	)
+	cancel := srv.Listen()
+	defer cancel()
+	origin := spec.ServerName(srv.ServerName())
+
+	fileName := "test.png"
+	contentType := "image/png"
+
+	uri := alice.UploadContent(t, data.LargePng, fileName, contentType)
+	_, mediaId := client.SplitMxc(uri)
+
+	fedReq := fclient.NewFederationRequest(
+		"GET",
+		origin,
+		"hs1",
+		"/_matrix/federation/v1/media/thumbnail/"+mediaId+"?method=crop&width=14&height=14&animated=false",
+	)
+
+	resp, err := srv.DoFederationRequest(context.Background(), t, deployment, fedReq)
+	if err != nil {
+		t.Fatalf("thumbnail request for uploaded file failed: %s", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("thumbnail request for uploaded file failed with status code: %v", resp.StatusCode)
+	}
+
+	resType := resp.Header.Get("Content-Type")
+	mimeType := strings.Split(resType, ";")[0]
+
+	if mimeType != "multipart/mixed" {
+		t.Fatalf("thumbnail request did not return multipart/mixed response, returned %s instead", mimeType)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	length := len(body)
+
+	contentLength, err := strconv.Atoi(resp.Header.Get("Content-Length"))
```

Comment:
Can we instead compare the received boddy to `test.png`? I don't think we necessarily should mandate that servers set `Content-Length` in the black box test suite.

Reply:
I think it probably makes sense to move the content-length check to the existing Synapse-specific tests for this endpoint: https://github.com/element-hq/synapse/blob/70b0e386032ed2f3ecf25fcf9a4b6c31335ffdc4/tests/federation/test_federation_media.py#L152

Reply:
Complement should definitely be checking `Content-Length` for all requests on all implementations - maybe it already is and was just missing a call to the federation thumbnail endpoint, I don't know.

Reply:
To expand: HTTP servers can either respond with `Content-Length` *or* chunked transfer encoding, and I don't think we should be mandating one or the other. We could do this by explicitly checking for the two cases, but what we actually care about is that the file was correctly transmitted (and it won't be if the `Content-Length` is wrong, unless golang HTTP client does something non-HTTP-spec compliant).